src/cmd/uds.go

```diff
@@ -27,11 +27,21 @@ var createCmd = &cobra.Command{
 	Args:    cobra.MaximumNArgs(1),
 	Short:   lang.CmdBundleCreateShort,
 	PreRun: func(cmd *cobra.Command, args []string) {
-		if len(args) > 0 && !zarfUtils.IsDir(args[0]) {
-			message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+		pathToBundleFile := ""
+		if len(args) > 0 {
+			if !zarfUtils.IsDir(args[0]) {
+				message.Fatalf(nil, "(%q) is not a valid path to a directory", args[0])
+			}
+			pathToBundleFile = filepath.Join(args[0])
 		}
-		if _, err := os.Stat(config.BundleYAML); len(args) == 0 && err != nil {
-			message.Fatalf(err, "%s not found in directory", config.BundleYAML)
+		// Handle .yaml or .yml
+		bundleYml := strings.Replace(config.BundleYAML, ".yaml", ".yml", 1)
```

Comment:
Thanks for that refactor. One more thing, looks like you can remove the `pathToBundleFile` variable entirely and replace it in those `if` statements with `args[0]`

Reply:
I don't believe I can do that. The way I wrote this code, the handle .yaml or .yml section can handle either file type, whether or not the user provided a directory to the create command. If I make the change you suggested and the user doesn't provide a directory, args[0] will error with index out of range. This is why, while checking if the user provided a directory and checking if its a valid directory I join it to the `pathToBundleFile` string. Using the `pathToBundleFile` string allows the following .yaml vs .yml checks to work in either situation without issue. Unless I'm overlooking something or misunderstanding the suggestion.

Reply:
You are def correct, thanks for that clarification!